sdmetrics/reports/single_table/quality_report.py

```diff
@@ -0,0 +1,100 @@
+"""Single table quality report."""
+
+import itertools
+import sys
+
+import numpy as np
+import pandas as pd
+import tqdm
+
+from sdmetrics.single_table import (
+    ContingencySimilarity, CorrelationSimilarity, KSComplement, TVComplement)
+
+
+class QualityReport():
+    """Single table quality report."""
+
+    METRICS = {
+        'Column Shapes': [KSComplement, TVComplement],
+        'Column Pair Trends': [CorrelationSimilarity, ContingencySimilarity],
+    }
+
+    _metric_results = {}
+
+    def generate(self, real_data, synthetic_data, metadata):
+        """Generate report.
+
+        Args:
+            real_data (pandas.DataFrame):
+                The real data.
+            synthetic_data (pandas.DataFrame):
+                The synthetic data.
+            metadata (dict):
+                The metadata, which contains each column's data type as well as relationships.
+        """
+        metrics = list(itertools.chain.from_iterable(self.METRICS.values()))
+
+        for metric in tqdm.tqdm(metrics, desc='Creating report:'):
+            self._metric_results[metric.__name__] = metric.compute_breakdown(
+                real_data, synthetic_data)
+
+        self._property_breakdown = {}
+        for prop, metrics in self.METRICS.items():
+            prop_scores = []
+            for metric in metrics:
+                score = np.mean(
+                    [
+                        breakdown['score'] for _, breakdown
+                        in self._metric_results[metric.__name__].items()
+                    ]
+                )
+                prop_scores.append(score)
+
+            self._property_breakdown[prop] = np.mean(prop_scores)
+
+        self._overall_quality_score = np.mean(list(self._property_breakdown.values()))
+
+        self._print_results()
+
+    def _print_results(self, out=sys.stdout):
```

Comment:
Also minor but we have been putting any method used by another method on top. Can we move this before `generate`? Or we can switch our convention. Idk if this rule has been consistent in `sdmetrics`